src/orange/BoundingBoxUtils.hh

```diff
@@ -221,6 +221,70 @@ inline bool encloses(BoundingBox<T> const& big, BoundingBox<T> const& small)
     });
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Calculate the distance to the inside of the bbox from a pos and dir.
+ *
+ * If there is no intersection, the result will be inf.
+ */
+template<class T, class U>
+inline U calc_dist_to_inside(BoundingBox<T> const& bbox,
+                             Array<U, 3> const& pos,
+                             Array<U, 3> const& dir)
+{
+    if (is_inside(bbox, pos))
```

Comment:
I think we might be better served to put this outside the function: I could imagine situations where we know *a priori* whether the point is inside/outside the box.

Reply:
Then I assume we would a variation of:
```suggestion
    CELER_ASSERT(is_inside(bbox, pos);
```